src/libs/Navigation/types.ts

```diff
@@ -0,0 +1,401 @@
+/* eslint-disable @typescript-eslint/naming-convention  */
+import {CommonActions, NavigationContainerRefWithCurrent, NavigationHelpers, NavigationState, NavigatorScreenParams, PartialRoute, Route} from '@react-navigation/native';
+import {ValueOf} from 'type-fest';
+import CONST from '@src/CONST';
+import NAVIGATORS from '@src/NAVIGATORS';
+import SCREENS from '@src/SCREENS';
+
+type NavigationRef = NavigationContainerRefWithCurrent<RootStackParamList>;
+
+type NavigationRoot = NavigationHelpers<RootStackParamList>;
+
+type GoBackAction = Extract<CommonActions.Action, {type: 'GO_BACK'}>;
+type ResetAction = Extract<CommonActions.Action, {type: 'RESET'}>;
+type SetParamsAction = Extract<CommonActions.Action, {type: 'SET_PARAMS'}>;
+
+type ActionNavigate = {
+    type: ValueOf<typeof CONST.NAVIGATION.ACTION_TYPE>;
+    payload: {
+        name?: string;
+        key?: string;
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        params?: any;
+        path?: string;
+        merge?: boolean;
+    };
+    source?: string;
+    target?: string;
+};
+
+type StackNavigationAction = GoBackAction | ResetAction | SetParamsAction | ActionNavigate | undefined;
+
+type NavigationStateRoute = NavigationState['routes'][number];
+type NavigationPartialRoute = PartialRoute<Route<string>>;
+type StateOrRoute = NavigationState | NavigationStateRoute | NavigationPartialRoute;
+
+type CentralPaneNavigatorParamList = {
+    [SCREENS.REPORT]: {
+        reportActionID: string;
+        reportID: string;
+    };
+};
+
+type SettingsNavigatorParamList = {
+    [SCREENS.SETTINGS.ROOT]: undefined;
+    Settings_Share_Code: undefined;
```

Comment:
NAB but I think now is as good a time as any to standardize on always using constants for screen names. I don't like how we have two patterns for defining screen names, as I think it causes confusion when adding new screens or when adding a new constant for a screen name.

Reply:
I would love to work on this, would you be able to create an issue for this so I can do it in a separate PR? @roryabraham